The discrete field of rational numbers (#1111)
UniMath/agda-unimath: src/ring-theory/trivial-rings.lagda.md

      ＝ x
        by left-unit-law-mul-Ring R x
```

> ### Invertible elements of nontrivial rings are different from zero

Reviewer: ```suggestion
### Invertible elements of nontrivial rings are not equal to zero
```

The suggested wording is clearer, as we tend to use the wording "establish a difference" when we speak of elements that are "apart", and apartness is a stronger condition on elements than nonequality.

You can read a little about apartness relations at the following link
- https://unimath.github.io/agda-unimath/foundation.apartness-relations.html

It's a useful concept in constructive algebra, which reminds me that the page is missing a reference to the book I believe the formalization follows, A Course in Constructive Algebra.

Reviewer: Regardless, this book may be of interest to you if you're not already familiar with it :)

Author: Thanks for the tip and the feedback. Maybe it'll give me idea for my next PR.

```agda
module _
  {l : Level} (R : Ring l) (H : is-nontrivial-Ring R) (x : type-Ring R)
  where

  is-nonzero-is-invertible-element-nontrivial-Ring :
    is-invertible-element-Ring R x → zero-Ring R ≠ x
  is-nonzero-is-invertible-element-nontrivial-Ring (y , P , _) K =
    H ( ( inv (left-zero-law-mul-Ring R y)) ∙
        ( ap (mul-Ring' R y) K) ∙
        ( P))

> ```

Author: With this wording for the header, shouldn't this return `x ≠ zero-Ring R` instead?

Author: Maybe this calls for some `nonzero-elements-rings` module some time in the future?

Reviewer: > With this wording for the header, shouldn't this return `x ≠ zero-Ring R` instead?

That's hardly important imo

Reviewer: > Maybe this calls for some `nonzero-elements-rings` module some time in the future?

It may be that it is more productive to consider elements that are apart from zero, but I don't know.

Author: Ok. This wording is more coherent with the current definition of division rings, so let's keep it this way for now.

      ＝ x
        by left-unit-law-mul-Ring R x
```

### Invertible elements of nontrivial rings are different from zero

```agda
module _
  {l : Level} (R : Ring l) (H : is-nontrivial-Ring R) (x : type-Ring R)
  where

  is-nonzero-is-invertible-element-nontrivial-Ring :
    is-invertible-element-Ring R x → zero-Ring R ≠ x
  is-nonzero-is-invertible-element-nontrivial-Ring (y , P , _) K =

>     H
>       ( ( inv (left-zero-law-mul-Ring R y)) ∙
>         ( ap (mul-Ring' R y) K) ∙
>         ( P))

Reviewer: ```suggestion
    H ( ( inv (left-zero-law-mul-Ring R y)) ∙
        ( ap (mul-Ring' R y) K) ∙
        ( P))
```
